coffee/lib/abstract-chosen.coffee

```diff
@@ -217,7 +218,7 @@ class AbstractChosen
       this.winnow_results_set_highlight()
 
   get_search_regex: (escaped_search_string) ->
-    regex_string = if @search_contains then escaped_search_string else "\\b#{escaped_search_string}\\w*\\b"
+    regex_string = if @search_contains then escaped_search_string else "(?:^|\\s)#{escaped_search_string}[^\\s]*"
```

Comment:
One of the things we can do here is just `s/?://` – make the the non-capturing group a capturing group instead.

This would allow the check in `winnow_results` to be a simple:

```coffeescript
startpos += 1 if search_match[1]
```

Which is nice, because it keeps us from having to do another regex `test` in the loop for potentially numerous matches.

And my further suggestion would be to delete the line from `winnow_results` altogether, and have `get_search_text` and `search_string_match` (which are related to each other both by relevance and also in-file proximity) look like:

```coffeescript
  get_search_regex: (escaped_search_string) ->
    regex_string = if @search_contains then escaped_search_string else "(^|\\s)#{escaped_search_string}[^\\s]*"
    regex_string = "^#{regex_string}" unless @enable_split_word_search or @search_contains
    regex_flag = if @case_sensitive_search then "" else "i"
    new RegExp(regex_string, regex_flag)

  search_string_match: (search_string, regex) ->
    match = regex.exec(search_string)
    match.index += 1 if !@search_contains && match?[1] # <--- do the potential munging here
    match
```

(I've implemented this change locally, and it passes all the tests.)

This way, the consuming code doesn't have to care about how the match is made, and making this change locally also helped me see that we weren't checking `@search_contains` for the match, which is unnecessary at the moment, but satisfies my OCD "completeness" sense.

(The reason it's unnecessary at the moment is that `get_search_text` for both the jquery and prototype implementations strips leading and trailing whitespace. If that ever changed, though, adding the parallel `@search_contains` check would be robust against that. Granted, that's probably not going to change, but like I said: OCD completeness.)

Reply:
Love it! Love all of it. :heart:

Updated in ffd5919 and 9d613cf (kept the history so everyone can follow the discussion if need be later on).